it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyControllerISpec.scala

```diff
@@ -119,49 +112,77 @@ trait CheckAgencyControllerISpec extends BaseISpec with SessionDataMissingSpec {
       }
     }
 
-    def aChoiceRedirectingToCheckAgencyStatus(businessTypeValue: String) = {
-      "redirect to /check-agency-status" in {
+    def aChoiceRedirectingToCheckAgencyStatus(businessTypeValue: String, isValidBusinessTypeIdentifier: Boolean) = {
+      s"redirect to /check-agency-status when valid else throw BadRequestException when businessTypeIdentifier: $businessTypeValue" in {
```

Comment:
I think you should have a single test for the `BadRequestException` case outside of this `aChoiceRedirectingToCheckAgencyStatus` function and remove the `isValidBusinessTypeIdentifier` and `testCasesBusinessTypeIdentifierInput` val.